OOP-Milestone:Game logic

IjayAbby opened the pull request:
In this milestone we are to consider the following:

- Suggested branch name game_logic.

- Add real logic for your game - get rid of all inline comments and mocked logic used in the previous milestone.

- As the main.rb file should be responsible for user interaction, this is the only(!) one place where puts/print and get.chomp can be used.

- Think about how you can encapsulate different parts of the game logic in classes. Do not create one class for everything.

- Read about the DRY rule and use it.

- Think about the differences between attr_readers/attr_writers and atrr_acessors. Use atrr_acessors only if you really need it.

- Remember about the single responsibility rule. Always.

- Remember to cover edge cases: think what player could try to do to cheat the game and make sure that your program does not allow to perform invalid action.

Div685 committed:
Add bin and lib Folder

- Add main file under bin folder
- Add .gitkeep file

IjayAbby committed:
rename main.rb to main

Div685 committed:
update Readme.md file

- update Gemfile for rubocop linter

Div685 committed:
update Rspec Linter

IjayAbby committed:
update author details
license
typing errors

IjayAbby committed:
add license link

IjayAbby committed:
add lib folder and move .gitkeep file inside it

Div685 committed:
Add Player method

- Add player X and print statement

IjayAbby committed:
add player2

IjayAbby committed:
add player to make file executable

IjayAbby committed:
add final newline

Div685 committed:
Add methods in main class

- Defined methods and comments for descriptions

Div685 committed:
update main and board file

- resolve rubocop worning
- display board with number in board file

Div685 committed:
Update main file for user interface

- Removed logic parts of class
- Add loop
- Add winning player
- Add different statement in loop

Div685 committed:
update main file

- Add Win move
- Add loop for player_turn

Div685 committed:
update valid_input method

- Add name1 and name1 in valid_input method

IjayAbby committed:
Add inline comments on methods

Div685 committed:
Add player class

- Add player class in separate file

IjayAbby committed:
add display_board method

Div685 committed:
Add choose_check_box method

- Add choice for player turn

IjayAbby committed:
add turn method

IjayAbby committed:
add rubocop disabler to player.rb file
add play_again? method

Div685 committed:
Add winner? method

- Add draw method

mahmoud717 reviewed (changes requested):
### Changes required ♻️ 

**Hello @IjayAbby  & @Div685**

Great job with the project so far, but there are still some changes to be made:
 
---
### Requested 🔴 

- [x] There are supposed to be no puts or gets outside of the main.rb file, please make the changes that your main file has the logic to run game and you lib files have the methods to store and save the logic.


---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

Div685 committed:
update main file

- Removed puts and gets from logic files
- Add puts and gets in the main.rb file

Temmarie reviewed (approved):
## Status: Approved :heavy_check_mark:

### Hi @Div685 and @IjayAbby ! 👋 👋 

### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: :tada:

### You may move on to the next milestone, good luck with your next projects! :rocket:
### Happy Coding! :+1:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

IjayAbby committed:
add play_again and quit_game methods
